src/mrpro/data/SpatialDimension.py

```diff
@@ -101,3 +102,136 @@ def zyx(self) -> tuple[T, T, T]:
     def __str__(self) -> str:
         """Return a string representation of the SpatialDimension."""
         return f'z={self.z}, y={self.y}, x={self.x}'
+
+    def __getitem__(
+        self: SpatialDimension[torch.Tensor], idx: typeing_utils.IndexerType
+    ) -> SpatialDimension[torch.Tensor]:
+        """Get SpatialDimension item."""
+        if not all(isinstance(el, torch.Tensor) for el in self.zyx):
+            raise IndexError('Cannot index SpatialDimension with non-tensor')
+        return SpatialDimension(self.z[idx], self.y[idx], self.x[idx])
+
+    def __setitem__(self: SpatialDimension[torch.Tensor], idx: typeing_utils.IndexerType, other: SpatialDimension):
+        """Set SpatialDimension item."""
+        if not all(isinstance(el, torch.Tensor) for el in self.zyx):
+            raise IndexError('Cannot index SpatialDimension with non-tensor')
+        self.z[idx] = other.z
+        self.y[idx] = other.y
+        self.x[idx] = other.x
+
+    @overload
+    def __mul__(self: SpatialDimension[int], other: int | SpatialDimension[int]) -> SpatialDimension[int]: ...
+    @overload
+    def __mul__(self: SpatialDimension[int], other: float | SpatialDimension[float]) -> SpatialDimension[float]: ...
+    @overload
+    def __mul__(self: SpatialDimension[float], other: float | SpatialDimension[float]) -> SpatialDimension[float]: ...
+
+    @overload
+    def __mul__(
+        self: SpatialDimension[torch.Tensor], other: float | T | SpatialDimension
+    ) -> SpatialDimension[torch.Tensor]: ...
+
+    def __mul__(self: SpatialDimension[T], other: T | float | SpatialDimension) -> SpatialDimension:
```

Comment:
see `float` / `T` comment above. If we do need it, using the same order everywhere would be nice. 

Please also check the rest of the code... appeards several times

Reply:
no, it is required, you can write it out as, for example:
SpatialDimension[int], other: int | float | SpatialDimension
All spacial dimensions can be multiplied by a float (which then includes int).
But only SpatialDimension[np.ndarray] can be multiplied by np.ndarray.

Reply:
i can go through and sort the order

Reply:
but T is `int | float | torch.Tensor` already, no?
so `float | T` is `float | int | float | torch.Tensor` , no?

Or am I actually missing something here?

Reply:
ah, got it... it's needed. Sorry for the confusion!

Reply:
but maybe your are right an the cases can be cleaned up, let me check